datafusion-postgres/src/datatypes.rs

```diff
@@ -18,6 +18,12 @@ use pgwire::error::{ErrorInfo, PgWireError, PgWireResult};
 use timezone::Tz;
 
 pub(crate) fn into_pg_type(df_type: &DataType) -> PgWireResult<Type> {
+    // Handle Dictionary types specially
+    if let DataType::Dictionary(_, value_type) = df_type {
+        // For Dictionary types, use the value type for mapping to Postgres types
+        return into_pg_type(value_type);
+    }
```

Comment:
I think we can write this into `match df_type` and it doesn't have to be an early return